quickstart/k8s-the-prod-way/architecture.md

```diff
@@ -0,0 +1,91 @@
+  is _always_ more time-efficient.
+* **Tractable plans.** It is markedly easier to notice that you're destroying your primary datastore
+  in a 30-line plan than a 900-line plan.
+* **Users get permissions they _need_ rather than global admin permissions.**
+* **Limited blast radius.** Separating (_e.g._) app and storage configration means that the app team
+  can't accidentally destroy the primary datastore.
+* **Independently managable.** For example, the identity stack (containing, _e.g._, GCP IAM) can
+  expose roles and credentials; the permissions of a specific role can be changed without making
+  changes to (say) the application layer.
+
+And so on.
+
+We advocate splitting infrastructure up into (roughly) 3 sets of resources. An explanation of the 3
+stacks follows the diagram. In the next section, we begin to provision and configure each of these
+stacks.
+
+<img src="/images/k8s-the-prod-way/kube-arch.png">
+
+## 1. Identity
+
+Identities and role definitions for organizations and CI/CD are required before anyone can provision
+anything. This is a requirement for every production Kubernetes deployment.
+
+One side-effect of isolating resources into loosely-coupled stacks is that we have the opportunity
+to grant **minimal permissions** based on need.
+
+Typically the identity stack looks something like this:
+
+1. **AWS/Azure/GCP identities and roles for the team.** (_e.g._, [AWS IAM][aws-iam], [GCP
+   IAM][gcp-iam], [Azure AD][azure-ad].) For example, the databases team typically gets _only_
+   administrative permissions for the datastores, while an app team might _only_ get cluster
+   developer permissions on for (_e.g._) GKE.
+1. **AWS/Azure/GCP service accounts for CI/CD.** While IAM roles and Active Directory accounts
+   describe identity of users, service accounts grant an identity for _workloads_, _e.g._, "Storage
+   CI/CD." The identity stack provisions these as well, with similarly minimal permissions. For
+   example, the storage infrastructure service account has _only_ administrative permissions, _etc_.
+
+## 2. Shared, managed infrastructure (_e.g._, compute, networking, storage)
+
+The next step is to provision all shared, managed infrastructure. At minimum, this typically this
+includes networking infrastructure and a managed Kubernetes platform (_e.g._, [EKS][eks],
+[GKE][gke], [AKS][aks]), and frequently also involves a managed datastore (_e.g._, [Aurora][aurora],
+[Cloud SQL][cloud-sql], [CosmosDB][cosmos-db]).
+
+Typically this also involves provisioning Kubernetes cluster infrastructure -- things like
+`Namespace`s, `ClusterRoleBinding`s, and so on.
+
+In (1), we provisioned a service account with admin permissions to each of the managed
+infrastructure offerings we wish to use. This is used in CI/CD. Typically all changes to shared,
+managed infrastructure are executed by this service account.
+
+## 3. Applications
+
```

Comment:
Missing from this picture is shared infrastructure services (data plane like Envoy, control plane like Istio and Consul, metrics services like Prometheus, etc). And then below, application services (logging, etc).

Reply:
This is a great opportunity to really showcase our ability to have batteries included cover the space of installing and managing these add on services. I think this is a huge win for us given the difficulty of doing it today. Even if this is a "coming soon" section, I'd really love to pencil them in.

Reply:
That's meant to be covered by "system components" but (a) I agree, and (b) I tried to add more stuff but couldn't make them fit because I am not good at this.

The plan is that Marc will make this suck less.